Extensions/BBText/bbtextruntimeobject-pixi-renderer.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * The PIXI.js renderer for the BBCode Text runtime object.
+ *
+ * @class BBTextRuntimeObjectPixiRenderer
+ * @constructor
+ * @param {gdjs.BBTextRuntimeObject} runtimeObject The object to render
+ * @param {gdjs.RuntimeScene} runtimeScene The gdjs.RuntimeScene in which the object is
+ */
+gdjs.BBTextRuntimeObjectPixiRenderer = function(runtimeObject, runtimeScene) {
+  this._object = runtimeObject;
+
+  // Load (or reset) the text
+  if (this._pixiObject === undefined) {
+    this._pixiObject = new MultiStyleText(runtimeObject._text, {
+      default: {
+        fontFamily: runtimeObject._family,
+        fontSize: `${runtimeObject._size}px`,
+        fill: runtimeObject._color,
+        tagStyle: ['[', ']'],
+        wordWrap: true,
+        wordWrapWidth: runtimeObject._wrappingWidth,
+        align: runtimeObject._align,
+      },
+    });
+
+    this._object.hidden = !runtimeObject._visible;
+  } else {
+    console.log('runtime obj', runtimeObject._wrappingWidth, this._pixiObject);
+    // this._pixiObject._texture.baseTexture.source.currentTime = 0;
+  }
+
+  runtimeScene
+    .getLayer('')
+    .getRenderer()
+    .addRendererObject(this._pixiObject, runtimeObject.getZOrder());
+
+  // Set the anchor in the center, so that the object rotates around
+  // its center
+  this._pixiObject.anchor.x = 0.5;
+  this._pixiObject.anchor.y = 0.5;
+
+  this.updatePosition();
+  this.updateAngle();
+  this.updateOpacity();
+  this.updateVisible();
+};
+
+gdjs.BBTextRuntimeObjectRenderer = gdjs.BBTextRuntimeObjectPixiRenderer;
+
+gdjs.BBTextRuntimeObjectPixiRenderer.prototype.getRendererObject = function() {
+  return this._pixiObject;
+};
+
+gdjs.BBTextRuntimeObjectPixiRenderer.prototype.ensureUpToDate = function() {
+  if (
+    !this._textureWasValid &&
+    this._pixiObject.texture &&
+    this._pixiObject.texture.valid
+  ) {
+    if (this._pixiObject._style.wordWrapWidth !== this._object._wrappingWidth) {
+      this._pixiObject._style.wordWrapWidth = this._object._wrappingWidth;
+      this._pixiObject.dirty = true;
+    }
+    if (this._object._align !== this._pixiObject._style.align) {
+      this._pixiObject._style.align = this._object._align;
+      this._pixiObject.dirty = true;
+    }
+    this.updatePosition();
+    this._textureWasValid = true;
+  }
+};
+
+/**
+ * Set the rendered width
+ * @param {number} width The new width, in pixels.
+ */
+gdjs.BBTextRuntimeObjectPixiRenderer.prototype.setWidth = function(width) {
+  this._object._wrappingWidth = width;
+  this._pixiObject._style.wordWrapWidth = this._object._wrappingWidth;
+  this._pixiObject.dirty = true;
+  this.updatePosition(); // Position needs to be updated, as position in the center of the PIXI Sprite.
+};
+
+gdjs.BBTextRuntimeObjectPixiRenderer.prototype.setBBText = function(text) {
+  this._pixiObject.setText(text);
+  this._pixiObject.updateText();
+  this.updatePosition();
+};
+
+gdjs.BBTextRuntimeObjectPixiRenderer.prototype.setBaseProperty = function(
```

Comment:
So this pattern works and allow to reduce a bit verbosity and share a few lines of code between these updates. While this is done in some library, it's still better to avoid this because:
* This reduces performance: every time you want to update the color, instead of paying for 1 function call, passing 1 string as parameter, you're paying for 1 function call, passing 2 strings as parameters and *then* you have a switch where you potentially have 5 string comparisons! (might be optimized by the JS engine, but not sure).
* It's hard to type this properly. What is the type of "value"? It depends on "property"! Adding comment with type annotations and getting this verified by TypeScript is hard
* It's error prone: what if I pass a string that is invalid like "coLor"? This goes unnoticed during the game :'( You could add an error message, but this could decrease the game performance. By using separate functions, you'll get a "hard error": the JS engine will throw an error and your game will stop - and when we'll have TypeScript, TypeScript will warn us about this (that could happen in case of typo).
* This reduce the ability to find method calls in the codebase (this is called sometime the ability to "grep"). Example: I want to search for all the calls to setBaseProperty with "color" in the codebase => hard to do! the string might be generated, coming from somewhere else - and searching "color" is so common that it's useless :'(
* It's also less easy to discover using autocompletion :)

Well, all of this to say: prefer avoiding this and make separate functions for each property :)